ntrans_gui.py

```diff
@@ -180,22 +208,42 @@ def generate_ntrans_dictionary(self) -> None:
             "target_language": self.target_language_var.get(),
         }
 
-        for key in self.user_choices:
-            if not self.user_choices[key]:
+        for key in user_choices:
+            if not user_choices[key]:
                 print(
                     "All fields need to be filled in before generating the N-Trans dictionary."
                 )  # TODO: Change print to label
                 break
 
         # TODO: Check filepath for validity
+        if not hasattr(ntrans_gui, "thread"):
```

Comment:
- Why are you referring to `ntrans_gui` instead of `self`? Those are the same thing, and it's more conventional to use `self` inside the class.
- It would be cleaner to add `self.thread = None` to `__init__` and replace this with `if self.thread is None`.

Reply:
I'm not sure `if self.thread is None` would work. self.thread is defined after this check, to not create duplicate threads. Which means that arriving here with `self.thread = None` in ` __init__` means that this check will never run the thread.

Reply:
You would change this check to `if self.thread is None`.